src/strands/experimental/bidi/models/novasonic.py

```
@@ -632,59 +562,29 @@ def _get_tool_result_event(self, content_name: str, result: dict[str, Any]) -> s
-            return
-
-        try:
-            async with self._send_lock:
-                for event in events:
-                    bytes_data = event.encode("utf-8")
-                    chunk = InvokeModelWithBidirectionalStreamInputChunk(
-                        value=BidirectionalInputPayloadPart(bytes_=bytes_data)
-                    )
-                    await self.stream.input_stream.send(chunk)
-                    logger.debug("nova sonic event sent successfully")
-
-        except Exception as e:
-            logger.error("error=<%s>, event=<%s> | error sending nova sonic event", e, event[:100])
-            raise
-
-    async def _initialize_client(self) -> None:
-        """Initialize Nova Sonic client."""
-        try:
-            config = Config(
-                endpoint_uri=f"https://bedrock-runtime.{self.region}.amazonaws.com",
-                region=self.region,
-                aws_credentials_identity_resolver=EnvironmentCredentialsResolver(),
-                auth_scheme_resolver=HTTPAuthSchemeResolver(),
-                auth_schemes={"aws.auth#sigv4": SigV4AuthScheme(service="bedrock")},
-            )
-
-            self.client = BedrockRuntimeClient(config=config)
-            logger.debug("region=<%s> | nova sonic client initialized", self.region)
-
-        except ImportError as e:
-            logger.error("error=<%s> | nova sonic dependencies not available", e)
-            raise
-        except Exception as e:
-            logger.error("error=<%s> | error initializing nova sonic client", e)
-            raise
+        async with self._send_lock:
+            for event in events:
+                bytes_data = event.encode("utf-8")
+                chunk = InvokeModelWithBidirectionalStreamInputChunk(
+                    value=BidirectionalInputPayloadPart(bytes_=bytes_data)
+                )
+                await self._stream.input_stream.send(chunk)
+                logger.debug("nova sonic event sent successfully")
```

Throughout this PR, I removed some try/excepts. We are going to let exceptions propagate upward and centralize the handling and logging in the BidiAgent class. This way, we can be more consistent across model providers.